atom/renderer/content_settings_client.cc

```diff
@@ -143,6 +143,36 @@ bool ContentSettingsClient::allowDatabase(const WebString& name,
   return allow;
 }
 
+
+void ContentSettingsClient::requestFileSystemAccessAsync(
+        const WebContentSettingCallbacks& callbacks) {
+  WebFrame* frame = render_frame()->GetWebFrame();
+  WebContentSettingCallbacks permissionCallbacks(callbacks);
+  if (frame->getSecurityOrigin().isUnique() ||
+      frame->top()->getSecurityOrigin().isUnique()) {
+      permissionCallbacks.doDeny();
+      return;
+  }
+
+  bool allow = true;
+  GURL secondary_url(
+      blink::WebStringToGURL(frame->getSecurityOrigin().toString()));
+  if (content_settings_manager_->content_settings()) {
+    allow =
+        content_settings_manager_->GetSetting(
+          GetOriginOrURL(frame),
+          secondary_url,
+          "cookies",
```

Comment:
ok, didn't think this was as closely related because Chrome handles it separately, but that's fine if it always makes sense together